Docs/pane styleguide (#1041)
siemens/ix: packages/documentation/docs/controls/_pane_styleguide.md

> Panes are interactive component that enable users to access hidden content or options that are not immediately visible on the screen. These drawers are usually hidden behind a button or icon and can be revealed by clicking on the control element. Panes can be used for organizing complex information, displaying multiple views or sections, collapsible or expandable, but not the solution for limited content, or confusing layout.

Reviewer: - component**s**
- Consistent wording: on Blind chapter it says "allow" instead of "enable"
- I would avoid using the term "drawer" here since this is a separate (though deprecated) component
- "confusing layout" => this is a general UX thing and not really connected to the panes

What I'm missing on the page is the actual awesome use cases and advantages, like:
- better task fokus -> related controls can be visually grouped, the main content is more tidied up
- more efficient use of screen space -> more compact and organized manner, especially for smaller screens
- more dynamic layouting option and visualizing hierarchy - than just everything in the main content
- gives contextual information -> often, panes contain contextual details e.g. about a selected option in the UI

Panes are interactive components that allow users to access hidden content or options that are not visible on the screen. 

It can help users to focus better on their tasks since related controls are visually grouped and the main content contains less information. They are beneficial to use for more compact and hierarchically better organized content, providing a more dynamic layout. You can use them to give contextual information or providing a more tidied up main content for your users for a better task focus.

![Pane overview](https://www.figma.com/file/wEptRgAezDU1z80Cn3eZ0o/iX-Pattern-Illustrations?type=design&node-id=1680-22044&mode=design&t=iP7h44Wf17P209P7-4)

1. Left pane (inline & expanded)
2. Top pane (inline)
3. Right pane (floating)
4. Bottom pane (inline)

> ## Variants

Reviewer: having a look at the button chapter - should here be better be the base construction described (collapse/expand button, icon, heading, content)?

> ## Variants

Reviewer: I'll move the illustration to the options section :)

Up to four floating or inline panes can be placed in a panel layout component.

- **Floating**: The panes are placed above the main content but below the nav menu and header.
- **Inline**: The panes are placed on one level with the main content. When expanded, they move the main content and reduce its available space.

## Options

- **Heading**: Set a headline for the pane. We typically use a short name that describes its content.
- **Icon**: An icon can be displayed on the header with the title.
- **Size**: Use one of the following values for the height of a left or right pane, and for the width of a top or bottom pane (on mobile, panes will always be full screen):
	- `240px`
	- `320px`
	- `360px`
	- `480px`
	- `33%`
	- `100%`
- **Borderless**: Panes have borders to visually split content areas. Use this option to hide them if panes are placed in other layouts than just the main content where borders are disruptive.

> - **Position**: Panes can be positioned on the left, top, right or bottom.

Reviewer: I'll check our examples again, good idea to give examples

- **Layout**: Depending on which pane should have more focus the top/bottom or left/right panes can use more space.

![Pane layouts](https://www.figma.com/file/wEptRgAezDU1z80Cn3eZ0o/iX-Pattern-Illustrations?type=design&node-id=1681-28910&mode=design&t=iP7h44Wf17P209P7-4)

1. Full height (left/right)
2. Full width (top/bottom)

## Behavior 

> - **Interaction**: The floating panes has a close button on the right side of the header; the inline pane has a collapse or expand button on the left side of the header.

Reviewer 2: ```suggestion
- **Interaction**: To expand floating panes, users typically click on a button or another interactive component within the main content. They can close floating panes by pressing on the button on the right side of the header which removed them from their view. Inline panes take a bar appearance when collapesd containing the title and the expand button. Users can expand inline panes by pressing on the expand button.
```
Another version: 
Floating panes have a close button on the right side of the header. Typically we place a button or another interactive element within the main content to trigger opening of floating panes. When expanded inline panes have a collapse button on the left side of the header. When collapsed, inline panes take a bar appearance containing the title and the expand button.

> - **Responsive**: On large and medium sized screens, all panes have a maximum width or height of `50%` of the screen. On small screens, all panes have full width and height (the header and nav menu is visible). Make sure your content is prepared for different screen sizes. Collapsed left and right panes are shown on the top and bottom for a more efficient use of space.

Reviewer 2: ```suggestion
- **Responsive**: On large and medium sized screens, all panes have a maximum width or height of `50%` of the screen. On small screens, all panes have full width and height (the header and nav menu is visible).  Collapsed left and right panes are shown on the top and bottom for a more efficient use of space.
```
Removed the "make sure" sentence as it was rather general and directive

![Pane small viewport](https://www.figma.com/file/wEptRgAezDU1z80Cn3eZ0o/iX-Pattern-Illustrations?type=design&node-id=1680-26548&mode=design&t=iP7h44Wf17P209P7-4)

1. Inline panes in collapsed state
2. Inline or floating pane in expanded state
3. Opened nav menu

> 

Reviewer 2: Could add "Stacking" behaviour here

## States

> Panes have two states: 

Reviewer 2: ```suggestion
Panes have two states: collapsed and expanded. Appearance of the states varies between variants and screen sizes.
```

> 1. Collapsed

Reviewer 2: ```suggestion

```

> 2. Expanded

Reviewer 2: ```suggestion
```

Floating panes (3) are not visible in the collapsed state since you trigger them from a control in the UI.

> ![Pane state](https://www.figma.com/file/wEptRgAezDU1z80Cn3eZ0o/iX-Pattern-Illustrations?type=design&node-id=1681-28456&mode=design&t=iP7h44Wf17P209P7-4)

Reviewer 2: As you have already a few illustrations, I would consider skipping this one.

## Dos and Don'ts

- Do use to organize the content when you have a lot. 
- Do use to display different views within the single screen. 
- Do use to expand or collapse the content to hide or reveal specific content. 
- Don't use for small amount of content.  

> 

Reviewer 2: Don't use for content that should always be visible to the user.
